src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs

```diff
@@ -68,10 +73,127 @@ public static string GetEffectiveGlobalPackagesFolder(string solutionDirectory,
 
         /// <summary>
         /// Create the lock file path from the config file path.
+        /// If the config file includes a project name the 
+        /// lock file will include the name also.
         /// </summary>
         public static string GetLockFilePath(string configFilePath)
         {
-            return Path.Combine(Path.GetDirectoryName(configFilePath), ProjectLockFileName);
+            string lockFilePath = null;
+
+            var dir = Path.GetDirectoryName(configFilePath);
+
+            var projectName = GetProjectNameFromConfigFileName(configFilePath);
+
+            if (projectName == null)
+            {
+                lockFilePath = Path.Combine(dir, ProjectLockFileName);
+            }
+            else
+            {
+                var lockFileWithProject = GetProjectLockFileNameWithProjectName(projectName);
+                lockFilePath = Path.Combine(dir, lockFileWithProject);
+            }
+
+            return lockFilePath;
+        }
+
+        /// <summary>
+        /// Creates a projectName.project.json file name.
+        /// </summary>
+        public static string GetProjectConfigWithProjectName(string projectName)
+        {
+            if (String.IsNullOrEmpty(projectName))
+            {
+                throw new ArgumentException(Strings.Argument_Cannot_Be_Null_Or_Empty);
+            }
+
+            return string.Format(CultureInfo.InvariantCulture, "{0}.{1}", projectName, ProjectConfigFileName);
+        }
+
+        /// <summary>
+        /// Creates a projectName.project.json file name.
+        /// </summary>
+        public static string GetProjectLockFileNameWithProjectName(string projectName)
+        {
+            if (String.IsNullOrEmpty(projectName))
+            {
+                throw new ArgumentException(Strings.Argument_Cannot_Be_Null_Or_Empty);
+            }
+
+            return string.Format(CultureInfo.InvariantCulture, "{0}.{1}", projectName, ProjectLockFileName);
+        }
+
+        /// <summary>
+        /// Parses a projectName.project.json file name into a project name.
+        /// If there is no project name null will be returned.
+        /// </summary>
+        public static string GetProjectNameFromConfigFileName(string configPath)
+        {
+            if (configPath == null)
+            {
+                throw new ArgumentNullException(nameof(configPath));
+            }
+
+            var file = Path.GetFileName(configPath);
+
+            string projectName = null;
+
+            if (file != null &&
+                IsProjectConfig(file) &&
+                file.Length >= (ProjectConfigFileName.Length + 1) &&
+                !string.Equals(file, ProjectConfigFileName, StringComparison.OrdinalIgnoreCase))
+            {
```

Comment:
Seems file.EndsWith(ProjectCofigFileEnding, StringComparison.OrdinalIgnoreCase) is good enough to replace the 3 conditions.